lib/services/device_info.dart

```diff
@@ -0,0 +1,28 @@
+import 'dart:io';
+import 'package:device_info/device_info.dart';
+import 'package:flutter/services.dart';
+
+class Deviceinfo {
+  String _identifier = '';
+  Future<String> getDeviceDetails() async {
```

Comment:
suggestion: The API is using the ID returned from this function to validate the user. Since this function is implying that it's possible for a user to have a blank ID, you'll likely end up with multiple users with that same ID. It would be safer for this function to either return null or generate a random UUID if the device ID can't be retrieved, and have the API return an error response if the ID parameter is null.